linera-views/src/backends/lru_caching.rs

```diff
@@ -27,28 +27,96 @@ use crate::{
 use crate::{memory::MemoryStore, store::TestKeyValueStore};
 
 #[cfg(with_metrics)]
-/// The total number of cache faults
-static NUM_CACHE_FAULT: LazyLock<IntCounterVec> =
-    LazyLock::new(|| register_int_counter_vec("num_cache_fault", "Number of cache faults", &[]));
+/// The total number of cache read value faults
+static NUM_CACHE_READ_VALUE_FAULT: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_read_value_fault",
+        "Number of cache read value faults",
+        &[],
+    )
+});
 
 #[cfg(with_metrics)]
-/// The total number of cache successes
-static NUM_CACHE_SUCCESS: LazyLock<IntCounterVec> =
-    LazyLock::new(|| register_int_counter_vec("num_cache_success", "Number of cache success", &[]));
+/// The total number of cache read value successes
+static NUM_CACHE_READ_VALUE_SUCCESS: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_read_value_success",
+        "Number of cache read value success",
+        &[],
+    )
+});
+
+#[cfg(with_metrics)]
+/// The total number of cache contains key faults
+static NUM_CACHE_CONTAINS_KEY_FAULT: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_contains_key_fault",
+        "Number of cache contains key faults",
+        &[],
+    )
+});
+
+#[cfg(with_metrics)]
+/// The total number of cache contains key successes
+static NUM_CACHE_CONTAINS_KEY_SUCCESS: LazyLock<IntCounterVec> = LazyLock::new(|| {
+    register_int_counter_vec(
+        "num_cache_contains_key_success",
+        "Number of cache contains key success",
+        &[],
+    )
+});
+
+enum CacheEntry {
+    DoesNotExist,
+    Exists,
+    Value(Vec<u8>),
+}
+
+impl CacheEntry {
```

Comment:
nit: To me, these functions are removing the benefits of using an enum `CacheEntry`. I would try to inline them.

Reply:
Ok, I inline the functions that are used only once.